dfastmi/io/CelerObject.py

```diff
@@ -37,36 +37,52 @@
     CelerProperties
 
 """
-from abc import ABC, abstractmethod
-from typing import List
+from abc import ABC
+from typing import List, Tuple
+from pydantic import BaseModel
+from dfastmi.io.AReach import AReach
 
-
-class ICelerObject(ABC):
+class ICelerObject(ABC, BaseModel):
     "Interface or abstract base class to the CelerObject."
-    @abstractmethod
-    def validate(self):
-        pass
-    
-
-
+    parent_reach : AReach = None
+   
 class CelerDischarge(ICelerObject):
-    cdisch = tuple[float,float]
-    
-    def validate(self):
-        if self.cdisch == (0.0, 0.0):            
-            # raise Exception(
-            #             'The parameter "CelerQ" must be specified for branch "{}", reach "{}" since "CelerForm" is set to 2.'.format(
-            #                 branch,
-            #                 reach,
-            #             )
-            #         )
-            return
+    """
+        Series of discharge, bed celerity pairs
+    """
+    cdisch : Tuple[float,float] = (0.0, 0.0)
     
+    def validate(self, values):
+        branch_name : str = self.parent_reach.parent_branch.name
+        reach_name : str = self.parent_reach.name
+        if self.cdisch == (0.0, 0.0):
+             raise ValueError(f'The parameter "CelerQ" must be specified for branch "{branch_name}", '
+                              f'reach "{reach_name}" since "CelerForm" is set to 2.')
```

Comment:
The information in this message does not correspond with the check that you're doing?

Reply:
This is the original warning. I believe it is correct. The default is (0.0, 0.0) if this is needed in the configuration because CelerForm is set to 2 and not available in the river configuration or available but set to this default we need to report/feedback this message.